common/include/JetCorrections.h

```diff
@@ -253,3 +254,89 @@ class JetResolutionSmearer: public uhh2::AnalysisModule {
     bool smear_met;
     int direction = 0; // -1 = down, +1 = up, 0 = nominal
 };
+
+////
+
+/** \brief generalization of JetResolutionSmearer (see the latter for additional info)
+ *         to apply jet-energy-resolution smearing on non-default jet collections
+ *
+ *  options parsed from Context:
+ *   - "jersmear_smear_met": if "true", propagate the jet resolution smearing to MET. Default is false.
+ *   - "jersmear_direction": either "nominal", "up", or "down" to apply nominal, +1sigma, -1sigma smearing correction
+ *
+ */
+class GenericJetResolutionSmearer : public uhh2::AnalysisModule {
+
+ public:
+  explicit GenericJetResolutionSmearer(uhh2::Context&, const std::string& recj="jets", const std::string& genj="genjets", const bool allow_met_smear=true);
+  virtual ~GenericJetResolutionSmearer() {}
+
+  virtual bool process(uhh2::Event&) override;
+
+  template<typename RJ, typename GJ> void apply_JER_smearing(std::vector<RJ>&, const std::vector<GJ>&, LorentzVector&);
+
+ private:
+  uhh2::Event::Handle<std::vector<Jet> >       h_recjets_;
+  uhh2::Event::Handle<std::vector<Particle> >  h_genjets_;
+  uhh2::Event::Handle<std::vector<TopJet> >    h_rectopjets_;
+  uhh2::Event::Handle<std::vector<GenTopJet> > h_gentopjets_;
+  bool smear_met;
+  int direction = 0; // -1 = down, +1 = up, 0 = nominal
+};
+
+template<typename RJ, typename GJ>
+void GenericJetResolutionSmearer::apply_JER_smearing(std::vector<RJ>& rec_jets, const std::vector<GJ>& gen_jets, LorentzVector& met){
+
+  // https://twiki.cern.ch/twiki/bin/view/CMS/JetResolution [13TeV JER measurement]
+  constexpr const size_t n = 7;
+
+  static float eta_hi[n] = {0.8  , 1.3  , 1.9  , 2.5  , 3.0  , 3.2  , 5.0  };
+  static float c_ct  [n] = {1.061, 1.088, 1.106, 1.126, 1.343, 1.303, 1.320};
+  static float c_up  [n] = {1.084, 1.117, 1.136, 1.220, 1.466, 1.414, 1.606};
+  static float c_dn  [n] = {1.038, 1.059, 1.076, 1.032, 1.220, 1.192, 1.034};
+
```

Comment:
Hi Marino, would it be possible to have these numbers only at one place in the code? At the moment, these lines are a duplication of what is given in the JetResolutionSmearer routine in the cxx file.